client/components/marketing-survey/gsuite-cancel-purchase-dialog/index.jsx

```diff
@@ -0,0 +1,144 @@
+ */
+import './style.scss';
+
+class GSuiteCancelPurchaseDialog extends Component {
+	state = {
+		isDisabled: false,
+		step: steps.resetSteps(),
+		surveyAnswerId: null,
+		surveyAnswerAdditionalText: '',
+	};
+
+	nextStepButtonClick = () => {
+		this.setState( ( { step } ) => {
+			return { step: steps.nextStep( step ) };
+		} );
+	};
+
+	previousStepButtonClick = () => {
+		this.setState( ( { step } ) => {
+			return { step: steps.previousStep( step ) };
+		} );
+	};
+
+	removeButtonClick = action => {
+		this.saveSurveyResults();
+		this.props.onRemovePurchase( action );
+	};
+
+	saveSurveyResults = async () => {
+		const { purchase, site, surveyAnswerId, surveyAnswerAdditionalText } = this.props;
+		const survey = wpcom.marketing().survey( 'calypso-gsuite-remove-purchase', purchase.siteId );
+		const surveyData = {
+			'why-cancel': {
+				response: surveyAnswerId,
+				text: surveyAnswerAdditionalText,
+			},
+			type: 'remove',
+		};
+		survey.addResponses( enrichedSurveyData( surveyData, moment(), site, purchase ) );
+
+		const response = await survey.submit();
+		if ( ! response.success ) {
+			notices.error( response.err );
+		}
+	};
+
+	onSurveryAnswerChange = ( surveyAnswerId, surveyAnswerAdditionalText ) => {
+		this.setState( {
+			surveyAnswerId,
+			surveyAnswerAdditionalText,
+		} );
+	};
+
+	getStepButtons = () => {
+		const { translate } = this.props;
+		const { step, isDisabled } = this.state;
+		if ( steps.GSUITE_INITIAL_STEP === step ) {
+			return [
+				{
+					action: 'cancel',
+					disabled: isDisabled,
+					label: translate( "I'll Keep It" ),
+				},
+				{
+					action: 'next',
+					disabled: isDisabled,
+					label: translate( 'Next Step' ),
+					onClick: this.nextStepButtonClick,
+				},
+			];
+		}
+		return [
+			{
+				action: 'cancel',
+				disabled: isDisabled,
+				label: translate( "I'll Keep It" ),
+			},
+			{
+				action: 'prev',
+				disabled: isDisabled,
+				label: translate( 'Previous Step' ),
+				onClick: this.previousStepButtonClick,
+			},
+			{
+				action: 'remove',
+				disabled: isDisabled,
+				isPrimary: true,
+				label: translate( 'Remove Now' ),
+				onClick: this.removeButtonClick,
+			},
+		];
+	};
+
+	render() {
+		const { isVisible, onClose, purchase } = this.props;
+		return (
+			<Dialog
+				buttons={ this.getStepButtons() }
+				className="gsuite-cancel-purchase-dialog"
+				isVisible={ isVisible }
+				onClose={ onClose }
+			>
+				{ steps.GSUITE_INITIAL_STEP === this.state.step ? (
+					<GSuiteCancellationFeatures purchase={ purchase } />
+				) : (
+					<GSuiteCancellationSurvey onSurveryAnswerChange={ this.onSurveryAnswerChange } />
+				) }
+			</Dialog>
+		);
+	}
+}
+
+GSuiteCancelPurchaseDialog.propTypes = {
+	isVisible: PropTypes.bool.isRequired,
+	onClose: PropTypes.func.isRequired,
+	onRemovePurchase: PropTypes.func.isRequired,
+	purchase: PropTypes.object.isRequired,
+	site: PropTypes.object.isRequired,
+	translate: PropTypes.func.isRequired,
+};
```

Comment:
I have been wavering between the two styles. Is it better to be consistent across functional vs class Components, or should the class ones have a static propTypes?

Reply:
My understanding is that we always embed proptypes in class components, and define them outside with functional ones. So I'd say it's probably better to stick with this convention for consistency purpose, unless of course this has changed - which I may very well not be aware of 🙂.

Reply:
I've seen it both ways many time.  If this is indeed the standard let's make the change.  Happy for that to happen in a quick follow up.

Reply:
I am going to make a point and ask, since with the advent of Hooks functional components will make more sense in more places, and defining the prototypes outside the components make it _much_ easier to refactor one into another.